libslide/src/bignum/complex.rs

```diff
@@ -0,0 +1,218 @@
+#![allow(clippy::suspicious_arithmetic_impl)]
+#![allow(dead_code)]
+use std::convert::TryFrom;
+use std::fmt;
+use std::ops;
+
+static INPUT_ERR_MSG: &str = "Input is not valid";
+const TOLERANCE: f64 = 1E-12;
+
+#[derive(Clone, Copy)]
+struct Complex {
+    real: f64,
+    imag: f64,
```

Comment:
Could we use Bignums here in the future?

Reply:
No, because then bignums multiplication would be circular. I intend to make a complex class for bignums in the future.